desktop/app/notifications.js

```diff
@@ -0,0 +1,38 @@
+import engine from '../../react-native/react/engine'
+import Window from './window'
+
+// Electron doesn't currently support showing notifications from the main
+// process, so we'll use a hidden window to show them for us.
+// https://github.com/atom/electron/issues/3359
+const window = new Window('notifier', { show: false })
+window.show()
```

Comment:
I can't remove all side effects without more work, for two reasons:

1. `listenOnConnect` currently needs to be called *before* the engine has connected, and the engine automatically connects when required. We could change it to immediately call a new `listenOnConnect` function if already connected.

2. The hidden window used to dispatch notifications needs to be created as early as possible, before any notifications are shown, but can't be created before `ready` fires on the app.

So to keep things simple, I'm going to leave this require where it is, but still wrap the guts of this guy in a function.